runtime/j9vm/jvm.c

```diff
@@ -1514,6 +1518,77 @@ static jint initializeReflectionGlobals(JNIEnv * env, BOOLEAN includeAccessors)
  *	DLL: jvm
  */
 jint JNICALL JNI_CreateJavaVM(JavaVM **pvm, void **penv, void *vm_args) {
+	return JNI_CreateJavaVM_impl(pvm, penv, vm_args, FALSE);
+}
+
+jint JNICALL JITServer_CreateServer(JITServer **jitServer, void *vm_args) {
+	JavaVM *jvm = NULL;
+	JNIEnv *env = NULL;
+	JITServer *server = NULL;
+	jint rc = JNI_OK;
+
+	server = malloc(sizeof(JITServer));
+	if (NULL == server) {
+		rc = JIT_SERVER_OOM;
+		goto _end;
+	}
+	*jitServer = server;
+	server->StartJITServer = StartJITServer;
+	server->WaitForTermination = WaitForTermination;
+	rc = JNI_CreateJavaVM_impl(&server->jvm, (void **)&env, vm_args, TRUE);
+
+	if (JNI_OK != rc) {
+		rc = JIT_SERVER_CREATE_FAILED;
+		goto _end;
+	}
+_end:
+	return rc;
+}
+
+jint JNICALL StartJITServer(JITServer *jitServer) {
+	JavaVM *vm = jitServer->jvm;
+	JNIEnv *env = NULL;
+	int32_t rc = 0;
+
+	rc = (*vm)->AttachCurrentThread(vm, (void **)&env, NULL);
```

Comment:
I am assuming that the attach operation is needed at some level. Is there a reason to detach immediately after `startJITServer`? I am asking because in the next operation (`WaitForTermination`) we again attach and then detach.

Reply:
It is not guaranteed that the user of `JITServer` would call `startJITServer` and `waitForTermination` on same thread. And from user perspective `JITServer` is not a JVM, but since we are internally creating a JVM we should hide the details that the creation and usage of JVM requires, like these attach and detach calls. Does that make sense?

Reply:
I am ok with the implementation in the JIT related files

Reply:
Looks like some how I managed to delete my answer to the first comment. For the record adding it again:
```It is possible that the calls to `startJITServer` and `WaitForTermination` are on different threads. Also the JITServer is not same as the JVM, but since we are internally creating a JVM, we should hide the details involved in creating and using JVM like these attach an detach calls.```